.github/workflows/linting.yml

```
@@ -0,0 +1,91 @@
+# workflow to run static-analysis and linting checks on source
+
+name: StaticAnalysis
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on pushes to the "main" branch and any pull request events
+  push:
+    branches: [ "main"]
+  pull_request:
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Workflow run - one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "static-analysis"
+  static-analysis:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m venv ftorch
+          . ftorch/bin/activate
+          pip install torch torchvision --index-url https://download.pytorch.org/whl/cpu
+          pip install fortitude-lint
+          pip install ruff
+
+      - name: FTorch CMake
+        run: |
+          . ftorch/bin/activate
+          export FT_DIR=$(pwd)
+          VN=$(python -c "import sys; print('.'.join(sys.version.split('.')[:2]))")
+          export Torch_DIR=${VIRTUAL_ENV}/lib/python${VN}/site-packages
```

```suggestion
          export Torch_DIR=$(pip show torch | grep Location | sed "s/Location: //")
```

one liner for you :) little simpler IMO

---

Discussed in person, since this is only on the GitHub runner and not to be used on other people's systems we agreed clarity was better than generality.